extensions/sql-migration/src/models/migrationWizardPage.ts

```diff
@@ -16,5 +16,37 @@ export abstract class MigrationWizardPage {
 
 	public abstract async onPageEnter(): Promise<void>;
 	public abstract async onPageLeave(): Promise<void>;
+
+	private readonly stateChanges: (() => Promise<void>)[] = [];
+	protected async onStateChangeEvent(e: StateChangeEvent) {
+
+		this.stateChanges.push((): Promise<void> => {
+			return this.handleStateChange(e);
+		});
+
+		this.enableQueueProcessor();
+	}
+
+	private queueActive = false;
+	private async enableQueueProcessor(): Promise<void> {
```

Comment:
Tests!

Reply:
I'll start writing these tests when we're certain we're keeping this wizard style system. There is a chance we're going to move to a dashboard style view for migrations, and that's going to have me changing a lot of the locations of these files and make some code irrelevant.